CONTRIBUTING.md

```diff
@@ -56,37 +55,188 @@ Please include the following with each issue:
+
+If any differences are in packages not modified on this branch, revert the changes before committing.
+
+### Add changelog entry
+
+`rush change`
+
+Follow prompts to enter a change description or press ENTER if the change does not warrant a changelog entry. If multiple packages have changed, multiple sets of prompts will be presented. If the changes are only to non-published packages (like the **test-apps**), then `rush change` will indicate that a changelog entry is not needed.
+
+Completing the `rush change` prompts will cause new changelog entry JSON files to be created.
+
+If using the command line, this process can be completed in one step by running `rushchange.bat` from the root directory.
+**Only use `rushchange.bat` if none of the changes require a changelog entry.**
+
+> Note: The CI build will break if changes are pushed without running `rush change` and `rush extract-api` (if the API was changed).
+
+Here is a sample [changelog](https://github.com/microsoft/rushstack/blob/master/apps/rush/CHANGELOG.md) to demonstrate the level of detail expected.
+
+---
+
+## Contributing Guidelines
+
+AppUI contains a large amount of legacy code. As such we are in the process of standardizing, reorganizing, and optimizing AppUI.
+
+Before contributing to AppUI please consult [the iTwinUI styling guide](https://github.com/iTwin/iTwinUI/blob/main/STYLEGUIDE.md)
+
+When contributing to AppUI please keep in mind the following guidelines.
+
+### Application Architecture
+
+Our `ui` folder contains all our packages which are organized as following:
+
+- `appui-react` should contain exclusively layouts. These are the largest and most complicated components in AppUI.
+- `components-react` should contain exclusively widgets: smaller components that are used inside `appui-react`. Ideally should largely be combinations of iTwinUI components.
+- `core-react` marked to be decrepit at some point. Currently, houses the smallest components inside AppUI. Ideally, all of these components would be replaced with the latest iTwinUI components and the entire core-react could be deleted. Rely on this as little as possible.
+- `imodel-components-react/` contains components that require a 3D environment. Used exclusively for iModels.
+
+### Common issues to solve
+
+Common fixes are we looking to solve include, but ar e not limited to:
+
+- Replacing `core-react` components with iTwinUI components
+- Deleting Snapshot tests
+- Migrating to functional from class based React Components
+
+### Conventions
+
+Please follow these conventions when contributing to AppUI
+
+- Be sure that your branch is up to date with the master branch (i.e. `git merge origin/master`)
+- Classnames should follow this formula: [abrv. of module name] + [folder] + [component]. For example, when working inside the Message Center module on MessageCenterField on the dialog, one would use `”mc-messagecenterfield-dialog”`
```

Comment:
```suggestion
- Classnames should follow this formula: [abbr. of package name] + [folders] + [component]. For example, when working on `MessageCenterField` component under `status-bar/fields` directory in `@itwin/appui-react` one would use `”uifw-statusBar-fields-dialog”`
```

Reply:
Let's add a note that we intend to move to CSS modules, until we do to minimize potential class name clashes we have this naming structure.

Let's list all abbreviations as well:
- `@itwin/appui-react` - `uifw`
- `@itwin/components-react` - `components`
- `@itwin/core-react` - `core`
- `@itwin/imodel-components-react` - `icr`